--- cmake/pcl_all_in_one_installer.cmake	Sun Jan 17 15:06:00 1971
+++ cmake/pcl_all_in_one_installer.cmake	Sun Jan 17 15:06:00 1971
@@ -24,17 +24,17 @@
 
     if(CMAKE_CL_64)
         set(OPENNI_PACKAGE "OpenNI-Win64-1.3.2-Dev.msi")
-        set(OPENNI_URL "http://dev.pointclouds.org/attachments/download/360/${OPENNI_PACKAGE}")
+        set(OPENNI_URL "http://www.pointclouds.org/assets/files/dependencies/${OPENNI_PACKAGE}")
         set(OPENNI_MD5 e54b5ede27b293c579c42dcede33b289)
         set(OPENNI_SENSOR_PACKAGE "Sensor-Win-OpenSource64-5.0.3.msi")
-        set(OPENNI_SENSOR_URL "http://dev.pointclouds.org/attachments/download/359/${OPENNI_SENSOR_PACKAGE}")
+        set(OPENNI_SENSOR_URL "http://www.pointclouds.org/assets/files/dependencies/${OPENNI_SENSOR_PACKAGE}")
         set(OPENNI_SENSOR_MD5 cbf4ce02d5ef430dca17833d5588e060)
     else(CMAKE_CL_64)
         set(OPENNI_PACKAGE "OpenNI-Win32-1.3.2-Dev.msi")
-        set(OPENNI_URL "http://dev.pointclouds.org/attachments/download/361/${OPENNI_PACKAGE}")
+        set(OPENNI_URL "http://www.pointclouds.org/assets/files/dependencies/${OPENNI_PACKAGE}")
         set(OPENNI_MD5 0b7118a0581abef411b58530d4039cf0)
         set(OPENNI_SENSOR_PACKAGE "Sensor-Win-OpenSource32-5.0.3.msi")
-        set(OPENNI_SENSOR_URL "http://dev.pointclouds.org/attachments/download/358/${OPENNI_SENSOR_PACKAGE}")
+        set(OPENNI_SENSOR_URL "http://www.pointclouds.org/assets/files/dependencies/${OPENNI_SENSOR_PACKAGE}")
         set(OPENNI_SENSOR_MD5 8bf14b2e813859f868fc316acb2d08fa)	
     endif(CMAKE_CL_64)
 
